server.py

```diff
@@ -22,43 +26,81 @@
 # Create state, will be edited
 state = get_start_state(map_name)
 
+card_pack = create_card_pack()
+card_pack = str(card_pack)
+robots = get_robot_names()
+
 # A list of connected clients
-websockets = []
+ws_receivers = []
+ws_interfaces = []
 # Routing table for aiohttp.web
 routes = web.RouteTableDef()
 
-@routes.get("/ws/")
-async def websocket(request):
+
+@routes.get("/receiver/")
+async def receiver(request):
     # Prepare WebSocket
     ws = web.WebSocketResponse()
     await ws.prepare(request)
-    # WebSocket is add to a list
-    websockets.append(ws)
+    # WebSocket is added to a list
+    ws_receivers.append(ws)
     try:
         # This message is sent only this (just connected) client
-        await ws.send_str(json.dumps(state.as_dict(map_name)))
+        # await ws.send_str(robots[0])
+        await ws.send_json(state.as_dict(map_name), dumps=json.dumps)
+        # await ws.send_str(card_pack)
 
-        #Process messages from this client
+        # Process messages from this client
         async for msg in ws:
             if msg.type == aiohttp.WSMsgType.TEXT:
                 print(msg.data)
 
                 # Send messages to all connected clients
-                for client in websockets:
+                for client in ws_receivers:
+                    # send info about state
+                    await client.send_str(json.dumps(state.as_dict(map_name)))
+        return ws
+    finally:
+            # after disconnection client is removed from list
+            ws_receivers.remove(ws)
+
+
+@routes.get("/interface/")
+async def interface(request):
+    # Prepare WebSocket
+    ws = web.WebSocketResponse()
+    await ws.prepare(request)
+    # WebSocket is added to a list
+    ws_interfaces.append(ws)
+    try:
+        # This message is sent only this (just connected) client
+        await ws.send_str(f"robot, {robots[0]}")
+        # await ws.send_json(state.as_dict(map_name), dumps=json.dumps)
+        await ws.send_str(f"cards, {card_pack}")
```

Comment:
Tady se nebude posílat card_pack, ale pouze dealt_cards.
 A je to divně naformátované, client obdrží: robot, hanka jakože tam je ta čárka.
A proč je tu robots[0]? Nechápu smysl této zprávy. Client interface obdrží celý stav, stejně jako client_receiver, poté bude teprve dostávat od serveru dealt_cards.

Reply:
Další přemýšlení...  a jo, fakt, tomuto klientu se nebude posílat stav, ale ty rozdané karty, bavili jsme se o tom, že bude pro sledování hry mít puštěné i to obecenstvo, takže stav uvidí tam že?

Reply:
Je to pokus o to, poslat ruzne druhy zprav a nejak potom rozlisit, ktera je ktera. Stringy jsou naformatovane tak, abych jen vedela, ze posila spravny druh dat - s robote jmeno robota, s kartama - karty. Zatim zde nemame logiku, ktera precte damages vybraneho robota a na zaklade toho mu posle odpovidajici pocet karet, tak cvicne pouzivam cely balicek. Neni to navrh k zacleneni :-)